### DISCUSSION TITLE
Server Error

### DISCUSSION THREAD

**Discussion by anjuugupta** [vercel/next.js#63893] [Help]
### Summary

please help me out

Server Error
Error: location is not defined

This error happened while generating the page. Any console logs will be displayed in the terminal window.
Call Stack
location
node_modules\next\dist\client\performance-relayer.js (20:20)
(rsc)/./node_modules/next/dist/client/performance-relayer.js
file:///D:/Desktop/E-Shop/e-shop/.next/server/vendor-chunks/next.js (2598:1)
__webpack_require__
file:///D:/Desktop/E-Shop/e-shop/.next/server/webpack-runtime.js (33:42)
require
node_modules\next\dist\client\index.js (51:69)
(rsc)/./node_modules/next/dist/client/index.js
file:///D:/Desktop/E-Shop/e-shop/.next/server/vendor-chunks/next.js (2543:1)
__webpack_require__
file:///D:/Desktop/E-Shop/e-shop/.next/server/webpack-runtime.js (33:42)
eval
(rsc)\node_modules\next\client.js (1:0)
(rsc)/./node_modules/next/client.js
file:///D:/Desktop/E-Shop/e-shop/.next/server/vendor-chunks/next.js (2291:1)
__webpack_require__
file:///D:/Desktop/E-Shop/e-shop/.next/server/webpack-runtime.js (33:42)
eval
webpack-internal:///(rsc)/./app/page.js (9:69)
(rsc)/./app/page.js
file:///D:/Desktop/E-Shop/e-shop/.next/server/app/page.js (261:1)
Function.__webpack_require__
file:///D:/Desktop/E-Shop/e-shop/.next/server/webpack-runtime.js (33:42)
runNextTicks
node:internal/process/task_queues (60:5)
listOnTimeout
node:internal/timers (540:9)
process.processTimers
node:internal/timers (514:7)
async eB
file:///D:/Desktop/E-Shop/e-shop/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js (35:401143)
async tt
file:///D:/Desktop/E-Shop/e-shop/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js (35:404857)
async tr
file:///D:/Desktop/E-Shop/e-shop/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js (35:405407)
async tr
file:///D:/Desktop/E-Shop/e-shop/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js (35:405538)
async tl
file:///D:/Desktop/E-Shop/e-shop/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js (36:2057)
async
file:///D:/Desktop/E-Shop/e-shop/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js (36:2596)

**Comment by SouthLink** [comment]
It looks like you're using the browser's API in a non-client API, because the page is not rendered yet,you just need to look like this:

before
```
const PageComponent = () => {
  const url = location.href;

  return (
    <Page />
  );
};

export default PageComponent;
```

after
```
import { useEffect, useState } from 'react';
  
const PageComponent = () => {
  const [url, setURL] = useState<string>('');

  useEffect(() => {
    setURL(location.href);
  }, [])
  
  return (
    <Page />
  );
};

export default PageComponent;
```
